src/main/java/spoon/reflect/visitor/CtBiScannerDefault.java

```diff
@@ -983,6 +983,20 @@ public void visitCtUsedService(spoon.reflect.declaration.CtUsedService usedServi
 		exit(usedService);
 	}
 
+	// autogenerated by CtBiScannerGenerator
+	@java.lang.Override
+	public void visitCtCompilationUnit(spoon.reflect.declaration.CtCompilationUnit compilationUnit) {
+		spoon.reflect.declaration.CtCompilationUnit other = ((spoon.reflect.declaration.CtCompilationUnit) (this.stack.peek()));
+		enter(compilationUnit);
+		biScan(spoon.reflect.path.CtRole.COMMENT, compilationUnit.getComments(), other.getComments());
+		biScan(spoon.reflect.path.CtRole.ANNOTATION, compilationUnit.getAnnotations(), other.getAnnotations());
+		biScan(spoon.reflect.path.CtRole.PACKAGE_DECLARATION, compilationUnit.getPackageDeclaration(), other.getPackageDeclaration());
```

Comment:
Why the package is not also a ref? When we discussed about the model architecture, the package got a CtRootPackage as ancestor, not a CtCompilationUnit.

Reply:
It is ref. Because CtCompilationUnits refers CtPackageDeclaration (new node type), which then has CtPackageReference.

Reply:
OK this is then used only if the CtCompilationUnit contains java types.
What if the CtCompilationUnit is a package-info or a module-info? Shouldn't we use inherited compilation units to manage those now?

Reply:
> OK this is then used only if the CtCompilationUnit contains java types.

I think it is used for package-info too. But I have not checked it deeper. It should behave same like before this PR.

Reply:
OK maybe we should clarify that in a future PR.